### PR TITLE
Correct machinectl become plugin function arguments

### PR DESCRIPTION
**Pull request by jb3**
##### SUMMARY

This pull request resolves a couple of bugs within the machinectl become plugin introduced when it was moved from the hardcoded setup to the modular setup with ansible 2.8.

- Changed a function call to `_get_option` to `get_option` to prevent an `AttributeError`.
- Changed an argument to `get_option` from `"flags"` to `"become_flags"` to prevent `ansible.errors.AnsibleError`.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
become

### CODE REVIEW

File: changelogs/fragments/58734-correct-machinectl-become-plugin.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+bugfixes:
+  - machinectl become plugin - call proper get_option method
+  - machinectl become plugin - correct configuration option
```
Comment (felixfontein) [inline changelogs/fragments/58734-correct-machinectl-become-plugin.yaml]: I think you can replace both lines with something like `  - machinectl become plugin - fix bugs which prevented plugin from working` :)

File: lib/ansible/plugins/become/machinectl.py [hunk 2]
```diff
@@ -81,7 +81,7 @@ def build_become_command(self, cmd, shell):
         if not cmd:
             return cmd
 
-        become = self._get_option('become_exe') or self.name
-        flags = self.get_option('flags') or ''
+        become = self.get_option('become_exe') or self.name
+        flags = self.get_option('become_flags') or ''
         user = self.get_option('become_user') or ''
-        return '%s shell -q %s %s@ -- %s' % (become, flags, user, cmd)
+        return '%s shell -q %s %s@ %s' % (become, flags, user, cmd)
```
Comment (felixfontein) [inline lib/ansible/plugins/become/machinectl.py]: This change was explicitly done in https://github.com/ansible/ansible/pull/56572

Is this change really necessary? Also, maybe it's depending on the machinectl version? @ttencate any opinion on this?

**Review by felixfontein — CHANGES_REQUESTED** [review]

File: lib/ansible/plugins/become/machinectl.py (hunk 2 above)
Comment (jb3) [reply lib/ansible/plugins/become/machinectl.py]: When running with systemd v232 (latest on Debian stretch) I get the following error when running with the `--` in the command:
```
fatal: [vps]: FAILED! => {
    "ansible_facts": {},
    "changed": false,
    "msg": "The following modules failed to execute: setup\n  setup: MODULE FAILURE\nSee stdout/stderr for the exact error\n"
}
```

and verbose logging displays the error as:
```
(1, b"Failed to get shell PTY: Specified path '--' is not absolute\r\n", b'Shared connection to <cactus ip> closed.\r\n')
```

When executing directly on the server as well I get the same issue: 
```
joseph@cactus:~$ sudo machinectl shell -q -M buildContainer root@ /bin/echo Hello
Hello
joseph@cactus:~$ sudo machinectl shell -q -M buildContainer root@ -- /bin/echo Hello
Failed to get shell PTY: Specified path '--' is not absolute
```

Comment (ttencate) [reply lib/ansible/plugins/become/machinectl.py]: I did some more checking on this, and it seems that my fix was wrong: https://github.com/ansible/ansible/issues/56571#issuecomment-508671679

I agree that this should be reverted for now. Could someone also reopen #56571 in that case?

Comment (jb3) [reply lib/ansible/plugins/become/machinectl.py]: Moved the `-q` argument to prepend the shell argument in the latest commit.

Comment (ttencate) [reply lib/ansible/plugins/become/machinectl.py]: Thanks! Note that that only fixes it for v230, not for versions < v230, so #56571 should still be reopened.

**Review by bcoca — APPROVED** [review]
